src/components/hv-date-field/index.js

```diff
@@ -42,6 +42,7 @@ import styles from './styles';
 export default class HvDateField extends PureComponent<Props, State> {
   static namespaceURI = Namespaces.HYPERVIEW;
   static localName = LOCAL_NAME.DATE_FIELD;
+  static localNameAliases = [];
```

Comment:
If an element has no aliases, is there a benefit in adding this static variable? Same for the rest of the elements.

Reply:
I originally added it for every components, as flow would not play nice with it's absence, even when the component type definition was:

```js
export type HvComponent<Props> = ComponentType<Props> & {
  localName: LocalName | string,
  localNameAliases?: ?Array<LocalName | string>,
  namespaceURI: NamespaceURI,
};
```

can you think of a way to make this optional while keeping flow happy?

Reply:
I wasn't sure why the property was added for each component, that's why I wanted to get it clarified. So the following is totally optional since the current approach also works.

This could be possibly due to [type variance](https://flow.org/en/docs/types/interfaces/#toc-interface-property-variance-read-only-and-write-only) since interface properties are invariant by default.

Making the property co-variant should solve this.

```diff
export type HvComponent<Props> = ComponentType<Props> & {
  localName: LocalName | string,
-  localNameAliases: Array<LocalName | string>,
+  +localNameAliases?: Array<LocalName | string>,
  namespaceURI: NamespaceURI,
};
```

Alternatively since the components themselves are not interested in the `localNameAliases` we can also extract all aliases to a common service or helper.

No strong feelings for any approaches.

Reply:
making the property covariant does not seem to help here, the errors I'm getting are:

```sh
Error: src/components/hv-date-field/stories/index.js:15
 15: const createStory = Helpers.stories(HvDateField);
                                         ^^^^^^^^^^^ statics of `HvDateField`. This type is incompatible with the expected param type of
                                                             v
304: export type HvComponent<Props> = ComponentType<Props> & {
305:   localName: LocalName | string,
306:   +localNameAliases?: Array<LocalName | string>,
307:   namespaceURI: NamespaceURI,
308: };
     ^ object type. See: src/types.js:304
  Property `localNameAliases` is incompatible:
                                                                 v
    304: export type HvComponent<Props> = ComponentType<Props> & {
    305:   localName: LocalName | string,
    306:   +localNameAliases?: Array<LocalName | string>,
    307:   namespaceURI: NamespaceURI,
    308: };
         ^ property `localNameAliases`. Property not found in. See: src/types.js:304
     15: const createStory = Helpers.stories(HvDateField);
                                             ^^^^^^^^^^^ statics of `HvDateField`
```

Reply:
Seems like some issue with incorrect inference. I don't have a solution to this currently.

Reply:
Will leave as is for the time being, we can revisit when we update flow or migrate to typescript